enhancements/microshift/etcd-as-transient-systemd-unit.md

```diff
@@ -0,0 +1,123 @@
+---
+title: etcd
+authors:
+  - dusk125
+  - hasbro17
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - TBD
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# etcd as a Transient systemd Unit
+
+## Summary
+
+The enhancement proposes moving etcd out of the MicroShift binary and into a transient systemd unit that is launched and managed by MicroShift.
+
+## Motivation
+
+Currently MicroShift [bundles upstream etcd](https://github.com/openshift/microshift/blob/f7f2260e4fbff61654f478fc149cb7052261f87f/go.mod#L21) into its binary and runs it as a goroutine. This results in etcd dependencies being coupled with MicroShift which makes it harder to maintain and upgrade etcd independently of the platform like it is done with a separate openshift/etcd repo in the Openshift Container Platform (OCP).
+
+This enhancement outlines an update to MicroShift that would decouple its deployment of etcd to bring it closer to how etcd is maintained with OCP.
+
+Changing the deployment of etcd, outlined below, would alleviate concerns about building etcd with a version of shared dependencies - mainly library-go - that etcd is not currently built against, and therefore, not tested against.
+
+Additionally moving the etcd deployment into its own process would make it easier to debug and process logs via journatlctl.
+
+### User Stories
+
+1. "As a developer, I want to maintain etcd versions independently of microshift dependencies."
+
+2. "As an admin, I want to observe and debug the etcd server logs separately from other microshift processes."
```

Comment:
We could also solve this by replacing the logger instance with one that includes a field for the component name, which would produce output similar to what we have for other components. Let's include that in the alternatives section.

Reply:
> We could also solve this by replacing the logger instance with one that includes a field for the component name, which would produce output similar to what we have for other components. Let's include that in the alternatives section.

I think you'll hit the same problem we have with the rest of the control plane, where a single library is called from multiple sources and when logging, you cannot distinguish which logical binary is having trouble.

Reply:
We might be able to both have and enjoy some cake... with the transient systemd unit. It looks like we can pipe the output to stdout/err, getting the etcd logs into the MicroShift logs while still being able to get just the etcd logs from journalctl.

Reply:
I don't think we need to do that. If we have the logs in systemd, they're already separated nicely. There's no particular reason to merge them back with the microshift logs.